openshift-with-appstudio-test/e2e/jvm_build_service_test.go

```diff
@@ -265,4 +264,59 @@ func TestExampleRun(t *testing.T) {
 			debugAndFailTest(ta, "timed out waiting for shaded-jdk11 to complete")
 		}
 	})
+
+	ta.t.Run("Correct JDK identified for JDK11 build", func(t *testing.T) {
+		//test that we don't attempt to use JDK8 on a JDK11+ project
+		err = wait.PollImmediate(ta.interval, 2*ta.timeout, func() (done bool, err error) {
+
+			dbList, err := jvmClient.JvmbuildserviceV1alpha1().DependencyBuilds(ta.ns).List(context.TODO(), metav1.ListOptions{})
+			if err != nil {
+				ta.Logf(fmt.Sprintf("error list dependencybuilds: %s", err.Error()))
+				return false, err
+			}
+			ta.Logf(fmt.Sprintf("number of dependencybuilds: %d", len(dbList.Items)))
+			for _, db := range dbList.Items {
+				if strings.Contains(db.Spec.ScmInfo.SCMURL, "shaded-jdk11") {
+					if db.Status.State == v1alpha1.DependencyBuildStateComplete ||
+						db.Status.State == v1alpha1.DependencyBuildStateBuilding ||
+						db.Status.State == v1alpha1.DependencyBuildStateSubmitBuild ||
+						db.Status.State == v1alpha1.DependencyBuildStateContaminated ||
+						db.Status.State == v1alpha1.DependencyBuildStateFailed {
```

Comment:
wouldn't it be easier to just say 
```
if scmurl.contains("shaded-jdk11") && state != new && state != analyzing {
  continue
}
``

otherwise do everthing else below, but with less indentation